mmv1/third_party/terraform/tests/resource_billing_budget_test.go

```diff
@@ -283,3 +283,132 @@ func TestBillingBudgetStateUpgradeV0(t *testing.T) {
 		})
 	}
 }
+
+func TestAccBillingBudget_budgetFilterProjectsOrdering(t *testing.T) {
+	t.Parallel()
+
+	context := map[string]interface{}{
+		"org":             getTestOrgFromEnv(t),
+		"billing_acct":    getTestBillingAccountFromEnv(t),
+		"random_suffix_1": randString(t, 10),
+		"random_suffix_2": randString(t, 10),
+	}
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckBillingBudgetDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccBillingBudget_budgetFilterProjectsOrdering1(context),
+				Check:  resource.ComposeTestCheckFunc(),
+			},
+			{
+				ResourceName:       "google_billing_budget.budget",
+				ImportState:        true,
+				ImportStateVerify:  true,
+				PlanOnly:           true,
+				ExpectNonEmptyPlan: false,
+			},
+
+			{
+				Config: testAccBillingBudget_budgetFilterProjectsOrdering2(context),
+			},
+			{
+				ResourceName:       "google_billing_budget.budget",
+				ImportState:        true,
+				ImportStateVerify:  true,
+				PlanOnly:           true,
```

Comment:
PlanOnly and ExpectNonEmptyPlan should both be on the same step as `Config: testAccBillingBudget_budgetFilterProjectsOrdering2(context),`.

That will say "Plan this config and assert that the plan must be empty".